src/inc/Crypto.hpp

```diff
@@ -3,15 +3,58 @@
 //  See LICENSE file in the project root for full license information.
 // 
 #pragma once
+#include "Exceptions.hpp"
 
+#include <memory>
+#include <string>
 #include <vector>
 
 namespace MSIX {
 
+    // Forward declaration of type defined within PAL
+    struct SHA256Context;
+
+    // Class used to compute SHA256 hashes over various sets of data.
+    // Create one and Add data to it if the data is not all available,
+    // or simply call ComputeHash if the data is all in memory.
     class SHA256
     {
     public:
-        static bool ComputeHash(std::uint8_t *buffer, std::uint32_t cbBuffer, std::vector<uint8_t>& hash);
+        using HashBuffer = std::vector<uint8_t>;
+
+        SHA256();
+
+        // Adds the next chunk of data to the hash.
+        void Add(const uint8_t* buffer, size_t cbBuffer);
+
+        inline void Add(const std::vector<std::uint8_t>& buffer)
+        {
+            Add(buffer.data(), buffer.size());
+        }
+
+        // Gets the hash of the data. This is a destructive action; the accumulated hash
+        // value will be returned and the object can no longer be used.
+        void Get(HashBuffer& hash);
+
+        inline HashBuffer SHA256::Get()

```

Comment:
>SHA256:: [](start = 26, length = 8)

Do we need it? I feel it looks weird

Reply:
Copy paste error that actually breaks in clang...

---
In reply to: [311784604](https://github.com/Microsoft/msix-packaging/pull/195#discussion_r311784604) [](ancestors = 311784604)